app/controllers/admin/tours_controller.rb

```diff
@@ -1,3 +1,63 @@
-class Admin::ToursController < ApplicationController
-  def index; end
+class Admin::ToursController < Admin::AdminController
+  before_action :prepare_create_tour, only: :create
+  before_action :find_tour_by_id, only: %i(update edit destroy)
+
+  def index
+    @pagy_tours, @tours = pagy Tour.newest
+  end
+
+  def new
+    @tour = Tour.new
+  end
+
+  def create
+    if @tour.save
+      flash[:success] = t "admin.tours.flash.created_tour_success"
+      redirect_to admin_tours_path
+    else
+      flash[:danger] = t "admin.tours.flash.created_tour_failed"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @tour.update tour_params.merge(category_id: params[:category_id])
+      flash[:success] = t "admin.tours.flash.updated_tour_success"
+      redirect_to admin_tours_path
+    else
+      flash[:danger] = t "admin.tours.flash.updated_tour_failed"
+      render :edit
+    end
+  end
+
+  def destroy
```

Comment:
tour mà xóa thì máy cái booking sao, mấy cái ng ta thanh toán sao thanh niên.  đang làm project chứ ko phải tutorial nữa nè

Reply:
cái này e định tạo được order thì sẽ check xem có order nào không thì mới thực hiện xóa ạ

Reply:
Em xử lý luôn đi

Reply:
còn comment nè, 
suggest, em tách hẵng 1 pull làm cái này, để ở đây là còn comment mãi

Reply:
vì những cái tour mà đã order nhưng mình ko còn thực hiện tour nữa thì sao